class.photon.php

```diff
@@ -454,21 +454,26 @@ public function filter_image_downsize( $image, $attachment_id, $size ) {
 
 				$photon_args = array();
 
-				// `full` is a special case in WP
-				// To ensure filter receives consistent data regardless of requested size, `$image_args` is overridden with dimensions of original image.
-				if ( 'full' == $size || ! $image_meta = image_get_intermediate_size( $attachment_id, $size ) ) {
+				$image_meta = image_get_intermediate_size( $attachment_id, $size );
+
+				if ( 'full' != $size && ! $image_meta ) {
```

Comment:
we can simplify these conditionals if the `if ( 'full' == size )` is first, then `elseif ( ! image_meta )`